java/src/jmri/jmrit/logixng/actions/ActionLightIntensity.java

```diff
@@ -0,0 +1,333 @@
+    public boolean isExternal() {
+        return true;
+    }
+    
+    @Override
+    public void vetoableChange(java.beans.PropertyChangeEvent evt) throws java.beans.PropertyVetoException {
+        if ("CanDelete".equals(evt.getPropertyName())) { // No I18N
+            if (evt.getOldValue() instanceof VariableLight) {
+                if (evt.getOldValue().equals(getLight().getBean())) {
+                    PropertyChangeEvent e = new PropertyChangeEvent(this, "DoNotDelete", null, null);
+                    throw new PropertyVetoException(Bundle.getMessage("LightIntensity_LightInUseVeto", getDisplayName()), e); // NOI18N
+                }
+            }
+        } else if ("DoDelete".equals(evt.getPropertyName())) { // No I18N
+            if (evt.getOldValue() instanceof VariableLight) {
+                if (evt.getOldValue().equals(getLight().getBean())) {
+                    removeLight();
+                }
+            }
+        }
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void execute() throws JmriException {
+        VariableLight light;
+        
+        switch (_addressing) {
+            case Direct:
+                light = _lightHandle != null ? _lightHandle.getBean() : null;
+                break;
+
+            case Reference:
+                String ref = ReferenceUtil.getReference(
+                        getConditionalNG().getSymbolTable(), _reference);
+                light = InstanceManager.getDefault(VariableLightManager.class)
+                        .getNamedBean(ref);
+                break;
+
+            case LocalVariable:
+                SymbolTable symbolTable = getConditionalNG().getSymbolTable();
+                light = InstanceManager.getDefault(VariableLightManager.class)
+                        .getNamedBean(TypeConversionUtil
+                                .convertToString(symbolTable.getValue(_localVariable), false));
+                break;
+
+            case Formula:
+                light = _expressionNode != null ?
+                        InstanceManager.getDefault(VariableLightManager.class)
+                                .getNamedBean(TypeConversionUtil
+                                        .convertToString(_expressionNode.calculate(
+                                                getConditionalNG().getSymbolTable()), false))
+                        : null;
+                break;
+
+            default:
+                throw new IllegalArgumentException("invalid _addressing state: " + _addressing.name());
+        }
+        
+        if (light == null) {
+//            log.warn("light is null");
+            return;
+        }
+        
+        double intensity = 0.0;
+        
+        if (_intensitySocket.isConnected()) {
+            intensity =
+                    ((MaleAnalogExpressionSocket)_intensitySocket.getConnectedSocket())
+                            .evaluate();
+        }
+        
+        light.setTargetIntensity(intensity);
```

Comment:
The valid range for intensity is 0.0 to 1.0.

Reply:
Fixed. Due to rounding errors, a formula might return -0.01 when 0.0 is expected, or 1.01 when 1.0 is expected. So the action now ensures that everything below 0.0 is set to 0.0 and everything above 1.0 is set to 1.0.